arm_instruction.c

```diff
@@ -28,10 +28,120 @@ Contact: Guillaume.Huard@imag.fr
+        break;
+    case 0x1:
+        return get_bit(zncv, Z)==0;
+        break;
+    case 0x2:
+        return get_bit(zncv, C)==1;
+        break;
+    case 0x3:
+        return get_bit(zncv, C)==0;
+        break;
+    case 0x4:
+        return get_bit(zncv, N)==1;
+        break;
+    case 0x5:
+        return get_bit(zncv, N)==0;
+        break;
+    case 0x6:
+        return get_bit(zncv, V)==1;
+        break;
+    case 0x7:
+        return get_bit(zncv, V)==0;
+        break;
+    case 0x8:
+        return (get_bit(zncv, C)==1 && get_bit(zncv, Z)==0);
+        break;
+    case 0x9:
+        return (get_bit(zncv, C)==0 || get_bit(zncv, Z)==1);
+        break;
+    case 0xA:
+        return (get_bit(zncv, N) == get_bit(zncv, V));
+        break;
+    case 0xB:
+        return (get_bit(zncv, N) != get_bit(zncv, V));
+        break;
+    case 0xC:
+        return get_bit(zncv, Z) ==0 && (get_bit(zncv, N)==get_bit(zncv, V));
+        break;
+    case 0xD:
+        return get_bit(zncv, Z) ==1 || (get_bit(zncv, N)!=get_bit(zncv, V));
+        break;
+    case 0xE:
+        return 1;
+    break;
+    }
+    return 0;
+}
+
+
+int arm_execute_instruction(arm_core p) {
+    uint32_t instruction;
+    arm_fetch(p, &instruction); 
+    if(!instruction_cond(p,instruction)) return 0; //verifier cond
+
+    switch (get_bits(instruction, 27, 25)) {
+        case 0b000:
+            if (get_bit(instruction, 4) && get_bit(instruction, 7)) { //5ieme instruction de la table
+                    return arm_load_store(p, instruction);
+                }
+                if (get_bits(instruction, 24, 23) == 2 && get_bit(instruction, 20) == 0) {
+                    if (get_bit(instruction, 4) == 0 || (get_bit(instruction, 4) == 1 && get_bit(instruction, 7) == 0)) {
+                    return arm_miscellaneous(p, instruction);
+                    }
+                }
+                else 
+                    return arm_data_processing_shift(p, instruction); //premiere instruction de la table
+        break;
+        case 0b001: 
+            switch(get_bits(instruction,24,20))
+    			{
+    				case 0b10010:
+                        break;
+    				case 0b10110:
+    					break;
+    				case 0b10000:
+                        break;
+    				case 0b10100: //undefined
+    					break;
+    				default:
+    					arm_data_processing_immediate_msr(p,instruction);
+    					break;
+    			}
+    	break;
+        case 0b010:
+            arm_load_store(p, instruction);
+        break;
+        case 0b011: 
+            if(get_bit(instruction, 4)==0)
+                arm_load_store(p, instruction);
+        break;
+        case 0b100: 
+            arm_load_store_multiple(p, instruction);
+        break;
+        case 0b101: 
+            arm_branch(p, instruction);
+        break;
+        case 0b110:
+            arm_coprocessor_load_store(p, instruction);
+        break;
+        case 0b111: 
+            arm_coprocessor_others_swi(p, instruction);
```

Comment:
software interrupt bit 24 must be 1, need to test for this bit

Reply:
Where do you find this info?

Reply:
![Screenshot_20231218-155508](https://github.com/CodeNachos/arm_simulator-1.5/assets/62766805/c4f5c047-7e34-4dad-b733-b01c3e42b610)

Reply:
At A3-2